libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/timeline/RustMatrixTimeline.kt

```diff
@@ -115,16 +114,54 @@ class RustMatrixTimeline(
                 postDiffs(diffs)
             }.launchIn(this)
 
-            innerTimeline.backPaginationStatusFlow()
+            paginationStateFlow()
                 .onEach {
-                    postPaginationStatus(it)
+                    _paginationState.value = it
                 }
                 .launchIn(this)
 
             fetchMembers()
         }
     }
 
+    private fun paginationStateFlow(): Flow<MatrixTimeline.PaginationState> {
+        return combine(
+            innerTimeline.backPaginationStatusFlow(),
+            timelineItems,
+        ) { paginationStatus, filteredItems ->
+            if (filteredItems.hasEncryptionHistoryBanner()) {
+                return@combine MatrixTimeline.PaginationState(
```

Comment:
I can observe that sometimes the EncryptionHistory banner was displayed and then some new Event were appearing (so it was back paginating). Is it correct that with this PR the app will not paginate as soon as the banner is present?

Reply:
Yes this is correct, thats the goal of the PR. It was creating weird behaviour in the poll history too.

Reply:
But the banner was displayed "by mistake", since actually there were some events when back paginating, so we introduce a bug here, no? The root cause of this may be fixed later, but I am afraid that the bug will be more visible.

Reply:
Yeah, we cant know for sure if there is more non encrypted events before. I fix the existing issue. If we want to change this behaviour, we should think about it with iOS and product.